queen-of-versions/src/main/java/com/infinum/queenofversions/GoogleInAppUpdateResumeSuccessListener.java

```diff
@@ -0,0 +1,59 @@
+package com.infinum.queenofversions;
+
+import android.app.Activity;
+import android.content.IntentSender;
+
+import com.google.android.play.core.appupdate.AppUpdateInfo;
+import com.google.android.play.core.appupdate.AppUpdateManager;
+import com.google.android.play.core.install.model.AppUpdateType;
+import com.google.android.play.core.install.model.InstallStatus;
+import com.google.android.play.core.install.model.UpdateAvailability;
+import com.google.android.play.core.tasks.OnSuccessListener;
+
+import co.infinum.princeofversions.UpdaterCallback;
+
+public class GoogleInAppUpdateResumeSuccessListener implements OnSuccessListener<AppUpdateInfo> {
+
+    private final int requestCode;
+    private final Activity activity;
+    private final AppUpdateManager appUpdateManager;
+    private final UpdaterCallback updaterCallback;
+    private final UpdaterStateCallback flexiableListener;
+    private final GoogleInAppUpdateFlexibleHandler handler;
+
+    GoogleInAppUpdateResumeSuccessListener(int requestCode, Activity activity, AppUpdateManager appUpdateManager,
+        UpdaterCallback updaterCallback, UpdaterStateCallback flexibleListener, GoogleInAppUpdateFlexibleHandler handler) {
+        this.requestCode = requestCode;
+        this.activity = activity;
+        this.appUpdateManager = appUpdateManager;
+        this.updaterCallback = updaterCallback;
+        this.flexiableListener = flexibleListener;
+        this.handler = handler;
+    }
+
+    @Override
+    public void onSuccess(AppUpdateInfo appUpdateInfo) {
+        if (appUpdateInfo.updateAvailability() == UpdateAvailability.DEVELOPER_TRIGGERED_UPDATE_IN_PROGRESS) {
+            updateInProgress(appUpdateInfo);
+        } else if (appUpdateInfo.installStatus() == InstallStatus.DOWNLOADED && appUpdateInfo.isUpdateTypeAllowed(AppUpdateType.FLEXIBLE) ) {
+            notifyUser();
+        }
+    }
+
+    private void updateInProgress(AppUpdateInfo appUpdateInfo) {
+        try {
+            appUpdateManager.startUpdateFlowForResult(
+                appUpdateInfo,
+                AppUpdateType.IMMEDIATE,
```

Comment:
In official documentation this DEVELOPER_TRIGGERED_UPDATE_IN_PROGRESS can only happen if the update is IMMEDIATE so thats why it's assumed that it will be always IMMEDIATE. But we could also add some check just to be sure.